lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -152,22 +195,22 @@ def add_shared_group_lines(lines, colorizer)
           lines
         end
 
-        def read_failed_line
+        def read_failed_lines
           matching_line = find_failed_line
           unless matching_line
-            return "Unable to find matching line from backtrace"
+            return ["Unable to find matching line from backtrace"]
           end
 
           file_path, line_number = matching_line.match(/(.+?):(\d+)(|:\d+)/)[1..2]
-
-          if File.exist?(file_path)
-            File.readlines(file_path)[line_number.to_i - 1] ||
-              "Unable to find matching line in #{file_path}"
-          else
-            "Unable to find #{file_path} to read failed line"
-          end
+          max_line_count = RSpec.configuration.max_displayed_failure_line_count
+          RSpec::Support.require_rspec_core "formatters/snippet_extractor"
```

Comment:
It'd be nice to find a way to not require this every time `read_failed_lines` runs.  I'm thinking the easiest solution might be the following:

* Move this require to the top of this file.
* Remove the require of this file from the top of `lib/rspec/core/notifications.rb`.
* Register `ExceptionPresenter` as an auto-loaded constant.

That way, the entire exception presentation sub-system is lazily loaded on demand via the first use of `ExceptionPresenter`.  When `rspec` runs and encounters no exceptions, it will not be loaded, but if there are any, it'll get loaded when RSpec is presenting the first failure.

Thoughts?